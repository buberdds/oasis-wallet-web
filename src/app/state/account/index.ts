import { PayloadAction } from '@reduxjs/toolkit'
import { Transaction } from 'app/state/transaction/types'
import { createSlice } from 'utils/@reduxjs/toolkit'
import { AccountState, Account } from './types'

export const initialState: AccountState = {
  address: '',
  liquid_balance: 0,

  accountError: null,
  transactions: [],
  transactionsError: null,
  loading: true,
}

const slice = createSlice({
  name: 'account',
  initialState,
  reducers: {
    clearAccount(state, action: PayloadAction<void>) {
      Object.assign(state, initialState)
    },
    fetchAccount(state, action: PayloadAction<string>) {},
    accountLoaded(state, action: PayloadAction<Account>) {
      state.accountError = null
      Object.assign(state, action.payload)
    },
    accountError(state, action: PayloadAction<string>) {
      state.accountError = action.payload
    },
    transactionsLoaded(state, action: PayloadAction<Transaction[]>) {
      state.transactionsError = null
      state.transactions = action.payload
    },
    transactionsError(state, action: PayloadAction<string>) {
      state.transactionsError = action.payload
    },
    setLoading(state, action: PayloadAction<boolean>) {
      state.loading = action.payload
    },
  },
})

export const { actions: accountActions } = slice

export default slice.reducer
