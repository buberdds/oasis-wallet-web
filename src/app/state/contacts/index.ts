import { PayloadAction } from '@reduxjs/toolkit'
import { createSlice } from 'utils/@reduxjs/toolkit'
import { ContactsState, Contact } from './types'

export const initialState: ContactsState = {}

const slice = createSlice({
  name: 'contacts',
  initialState,
  reducers: {
    add(state, action: PayloadAction<Contact>) {
      state[action.payload.address] = action.payload
    },
    update(state, action: PayloadAction<Contact>) {
      state[action.payload.address] = action.payload
    },
    delete(state, action: PayloadAction<string>) {
      delete state[action.payload]
    },
  },
})

export const { actions: contactsActions } = slice

export default slice.reducer
