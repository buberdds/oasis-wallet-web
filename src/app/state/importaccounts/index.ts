import { PayloadAction } from '@reduxjs/toolkit'
import { ErrorPayload } from 'types/errors'
import { createSlice } from 'utils/@reduxjs/toolkit'
import { ImportAccountsListAccount, ImportAccountsState, ImportAccountsStep } from './types'

export const initialState: ImportAccountsState = {
  accounts: [],
  deviceRequested: false,
  showAccountsSelectionModal: false,
}

const slice = createSlice({
  name: 'importAccounts',
  initialState,
  reducers: {
    clear(state, action: PayloadAction<void>) {
      state.accounts = []
      state.error = undefined
      state.step = undefined
      state.showAccountsSelectionModal = false
    },
    deviceRequested(state, action: PayloadAction<void>) {
      state.deviceRequested = true
    },
    enumerateAccountsFromLedger(state, action: PayloadAction<void>) {
      state.step = undefined
      state.accounts = []
      state.showAccountsSelectionModal = true
    },
    enumerateAccountsFromMnemonic(state, action: PayloadAction<string>) {
      state.step = undefined
      state.accounts = []
      state.showAccountsSelectionModal = true
    },
    toggleAccount(state, action: PayloadAction<number>) {
      const index = action.payload
      state.accounts[index].selected = !state.accounts[index].selected
    },
    accountsListed(state, action: PayloadAction<ImportAccountsListAccount[]>) {
      state.accounts = action.payload
    },
    setStep(state, action: PayloadAction<ImportAccountsStep>) {
      state.step = action.payload
    },
    operationFailed(state, action: PayloadAction<ErrorPayload>) {
      state.error = action.payload
      state.step = undefined
    },
  },
})

export const { actions: importAccountsActions } = slice

export default slice.reducer
