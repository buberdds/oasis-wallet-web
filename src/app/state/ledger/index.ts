import { PayloadAction } from '@reduxjs/toolkit'
import { ErrorPayload } from 'types/errors'
import { createSlice } from 'utils/@reduxjs/toolkit'
import { LedgerAccount, LedgerState, LedgerStep } from './types'

export const initialState: LedgerState = { accounts: [] }

const slice = createSlice({
  name: 'ledger',
  initialState,
  reducers: {
    clear(state, action: PayloadAction<void>) {
      state.accounts = []
      state.error = undefined
    },
    enumerateAccounts(state, action: PayloadAction<void>) {
      state.step = undefined
      state.accounts = []
    },
    toggleAccount(state, action: PayloadAction<number>) {
      const index = action.payload
      state.accounts[index].selected = !state.accounts[index].selected
    },
    accountsListed(state, action: PayloadAction<LedgerAccount[]>) {
      state.accounts = action.payload
    },
    setStep(state, action: PayloadAction<LedgerStep>) {
      state.step = action.payload
    },
    operationFailed(state, action: PayloadAction<ErrorPayload>) {
      state.error = action.payload
      state.step = undefined
    },
  },
})

export const { actions: ledgerActions } = slice

export default slice.reducer
