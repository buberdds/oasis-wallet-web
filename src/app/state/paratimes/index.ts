import { PayloadAction } from '@reduxjs/toolkit'
import { StringifiedBigInt } from 'types/StringifiedBigInt'
import { createSlice } from 'utils/@reduxjs/toolkit'
import { ParaTimesState, TransactionForm, TransactionFormSteps, TransactionTypes } from './types'

export const initialState: ParaTimesState = {
  balance: '',
  isLoading: false,
  transactionForm: {
    amount: '',
    confirmation: false,
    paraTime: undefined,
    privateKey: '',
    recipient: '',
    type: undefined,
  },
  transactionFormStep: TransactionFormSteps.TransferType,
}

const slice = createSlice({
  name: 'paraTimes',
  initialState,
  reducers: {
    balanceLoaded(state, action: PayloadAction<StringifiedBigInt>) {
      state.balance = action.payload
      state.isLoading = false
    },
    fetchBalanceUsingEthPrivateKey(state, action: PayloadAction<void>) {
      state.isLoading = true
    },
    fetchBalanceUsingOasisAddress(state, action: PayloadAction<void>) {
      state.isLoading = true
    },
    resetTransactionForm(state, action: PayloadAction<void>) {
      state.transactionForm = initialState.transactionForm
      state.transactionFormStep = TransactionFormSteps.TransferType
    },
    navigateToDeposit(state, action: PayloadAction<void>) {
      state.transactionForm.type = TransactionTypes.Deposit
      state.transactionFormStep = TransactionFormSteps.ParaTimeSelection
    },
    navigateToWithdraw(state, action: PayloadAction<void>) {
      state.transactionForm.type = TransactionTypes.Withdraw
      state.transactionFormStep = TransactionFormSteps.ParaTimeSelection
    },
    navigateToRecipient(state, action: PayloadAction<void>) {
      state.transactionFormStep = TransactionFormSteps.TransactionRecipient
    },
    navigateToAmount(state, action: PayloadAction<void>) {
      state.transactionFormStep = TransactionFormSteps.TransactionAmount
    },
    navigateToParaTimes(state, action: PayloadAction<void>) {
      state.transactionFormStep = TransactionFormSteps.TransferType
    },
    navigateToConfirmation(state, action: PayloadAction<void>) {
      state.transactionFormStep = TransactionFormSteps.TransactionConfirmation
    },
    navigateToSummary(state, action: PayloadAction<void>) {
      state.transactionFormStep = TransactionFormSteps.TransactionSummary
    },
    setTransactionForm(state, action: PayloadAction<TransactionForm>) {
      state.transactionForm = action.payload
    },
  },
})

export const { actions: paraTimesActions } = slice

export default slice.reducer
