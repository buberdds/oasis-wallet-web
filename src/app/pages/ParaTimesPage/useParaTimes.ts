import { useCallback } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { useTranslation } from 'react-i18next'
import { paraTimesActions } from 'app/state/paratimes'
import { TransactionForm, TransactionTypes } from 'app/state/paratimes/types'
import { selectSelectedNetwork, selectTicker } from 'app/state/network/selectors'
import { selectAccountAvailableBalance, selectAccountIsLoading } from 'app/state/account/selectors'
import { selectAddress } from 'app/state/wallet/selectors'
import { selectParaTimes } from 'app/state/paratimes/selectors'
import { paraTimesConfig, RuntimeTypes, ParaTime } from '../../../config'

type AvailableParaTimesForNetwork = {
  isEvm: boolean
  value: ParaTime
}

const evmcParaTimes = Object.keys(paraTimesConfig).filter(
  key => paraTimesConfig[key as ParaTime].type === RuntimeTypes.Evm,
)

export const useParaTimes = () => {
  const { t } = useTranslation()
  const dispatch = useDispatch()
  const resetTransactionForm = useCallback(() => {
    dispatch(paraTimesActions.resetTransactionForm())
  }, [dispatch])
  const setTransactionForm = (formValues: TransactionForm) =>
    dispatch(paraTimesActions.setTransactionForm(formValues))
  const { balance, isLoading, transactionForm } = useSelector(selectParaTimes)
  const accountBalance = useSelector(selectAccountAvailableBalance)
  const accountIsLoading = useSelector(selectAccountIsLoading)
  const accountAddress = useSelector(selectAddress)
  const selectedNetwork = useSelector(selectSelectedNetwork)
  const ticker = useSelector(selectTicker)
  const isDepositing = transactionForm.type !== TransactionTypes.Withdraw
  const isEvmcParaTime = evmcParaTimes.includes(transactionForm.paraTime!)
  const needsEthAddress = isDepositing && isEvmcParaTime
  const balanceInBaseUnit = isDepositing || (!isDepositing && !isEvmcParaTime)
  const paraTimeName = transactionForm.paraTime ? t(`paraTimes.common.${transactionForm.paraTime}`) : ''
  const availableParaTimesForSelectedNetwork: AvailableParaTimesForNetwork[] = (
    Object.keys(paraTimesConfig) as ParaTime[]
  )
    .filter(paraTimeKey => paraTimesConfig[paraTimeKey][selectedNetwork].runtimeId)
    .map(paraTimeKey => ({
      isEvm: paraTimesConfig[paraTimeKey].type === RuntimeTypes.Evm,
      value: paraTimeKey,
    }))
  const walletBalance = !isDepositing ? balance : accountBalance

  return {
    accountAddress,
    accountIsLoading,
    availableParaTimesForSelectedNetwork,
    balance: walletBalance,
    balanceInBaseUnit,
    isDepositing,
    isWalletEmpty: walletBalance === '0',
    isEvmcParaTime,
    isLoading,
    paraTimeName,
    resetTransactionForm,
    ticker,
    transactionForm,
    setTransactionForm,
    usesOasisAddress: !needsEthAddress,
  }
}
