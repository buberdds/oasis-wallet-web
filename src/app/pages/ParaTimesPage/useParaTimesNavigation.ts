import { useCallback } from 'react'
import { useDispatch } from 'react-redux'
import { paraTimesActions } from 'app/state/paratimes'

export const useParaTimesNavigation = () => {
  const dispatch = useDispatch()

  const navigateToDeposit = useCallback(() => dispatch(paraTimesActions.navigateToDeposit()), [dispatch])
  const navigateToWithdraw = useCallback(() => dispatch(paraTimesActions.navigateToWithdraw()), [dispatch])
  const navigateToParaTimes = useCallback(() => dispatch(paraTimesActions.navigateToParaTimes()), [dispatch])
  const navigateToRecipient = useCallback(() => dispatch(paraTimesActions.navigateToRecipient()), [dispatch])
  const navigateToAmount = useCallback(() => dispatch(paraTimesActions.navigateToAmount()), [dispatch])
  const navigateToConfirmation = useCallback(
    () => dispatch(paraTimesActions.navigateToConfirmation()),
    [dispatch],
  )
  const navigateToSummary = useCallback(() => dispatch(paraTimesActions.navigateToSummary()), [dispatch])

  return {
    navigateToAmount,
    navigateToDeposit,
    navigateToParaTimes,
    navigateToRecipient,
    navigateToConfirmation,
    navigateToSummary,
    navigateToWithdraw,
  }
}
