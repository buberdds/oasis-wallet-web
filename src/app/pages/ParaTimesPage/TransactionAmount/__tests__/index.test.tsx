import * as React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { paraTimesActions } from 'app/state/paratimes'
import { useParaTimes, ParaTimesHook } from '../../useParaTimes'
import { useParaTimesNavigation, ParaTimesNavigationHook } from '../../useParaTimesNavigation'
import { TransactionAmount } from '..'
import { ParaTime } from 'config'

const mockDispatch = jest.fn()
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch,
}))
jest.unmock('react-i18next')
jest.mock('../../useParaTimes')
jest.mock('../../useParaTimesNavigation')

describe('<TransactionAmount />', () => {
  const mockUseParaTimesResult = {
    balance: '1000000000',
    balanceInBaseUnit: true,
    isDepositing: true,
    isEvmcParaTime: false,
    isWalletEmpty: false,
    paraTimeName: 'Cipher',
    ticker: 'ROSE',
    transactionForm: {
      amount: '',
      recipient: 'dummyAddress',
      paraTime: ParaTime.Cipher,
      privateKey: 'privateKey',
    },
    usesOasisAddress: true,
  } as ParaTimesHook
  const mockUseParaTimesNavigationResult = {} as ParaTimesNavigationHook

  beforeEach(() => {
    jest.mocked(useParaTimes).mockReturnValue(mockUseParaTimesResult)
    jest.mocked(useParaTimesNavigation).mockReturnValue(mockUseParaTimesNavigationResult)
  })

  it('should render component', () => {
    const { container } = render(<TransactionAmount />)

    expect(container).toMatchSnapshot()
  })

  it('should render EVMc withdraw variant component', () => {
    jest.mocked(useParaTimes).mockReturnValue({
      ...mockUseParaTimesResult,
      isDepositing: false,
      isEvmcParaTime: true,
      paraTimeName: 'Emerald',
    } as ParaTimesHook)
    render(<TransactionAmount />)

    expect(screen.getByTestId('paraTime-content-description')).toMatchSnapshot()
  })

  it('should render disabled variant', () => {
    jest.mocked(useParaTimes).mockReturnValue({
      ...mockUseParaTimesResult,
      balance: '0',
      isWalletEmpty: true,
    } as ParaTimesHook)
    render(<TransactionAmount />)

    expect(screen.getByText('The wallet is empty. There is nothing to withdraw.')).toBeInTheDocument()
    expect(screen.getByPlaceholderText('0')).toBeDisabled()
    expect(screen.getByRole('button', { name: 'MAX' })).toBeDisabled()
    expect(screen.getByRole('button', { name: 'Next' })).toBeDisabled()
  })

  it('should set input value to max', () => {
    const setTransactionForm = jest.fn()
    jest.mocked(useParaTimes).mockReturnValue({ ...mockUseParaTimesResult, setTransactionForm })
    const { rerender } = render(<TransactionAmount />)

    userEvent.click(screen.getByRole('button', { name: 'MAX' }))
    expect(setTransactionForm).toHaveBeenNthCalledWith(1, expect.objectContaining({ amount: '1' }))

    jest
      .mocked(useParaTimes)
      .mockReturnValue({ ...mockUseParaTimesResult, setTransactionForm, balance: '2493300888889' })
    rerender(<TransactionAmount />)

    userEvent.click(screen.getByRole('button', { name: 'MAX' }))
    expect(setTransactionForm).toHaveBeenNthCalledWith(
      2,
      expect.objectContaining({ amount: '2,493.300888889' }),
    )
  })

  it('should require amount field on form submit', () => {
    const navigateToConfirmation = jest.fn()
    jest.mocked(useParaTimesNavigation).mockReturnValue({
      ...mockUseParaTimesNavigationResult,
      navigateToConfirmation,
    })
    render(<TransactionAmount />)

    userEvent.click(screen.getByRole('button', { name: 'Next' }))

    expect(screen.getByText('Field is required')).toBeInTheDocument()
    expect(navigateToConfirmation).not.toHaveBeenCalled()
  })

  it('should validate amount field on form submit', () => {
    const navigateToConfirmation = jest.fn()
    jest.mocked(useParaTimes).mockReturnValue({
      ...mockUseParaTimesResult,
      transactionForm: {
        amount: '10000000000000',
      },
    } as ParaTimesHook)
    jest.mocked(useParaTimesNavigation).mockReturnValue({
      ...mockUseParaTimesNavigationResult,
      navigateToConfirmation,
    })
    render(<TransactionAmount />)

    userEvent.click(screen.getByRole('button', { name: 'Next' }))

    expect(screen.getByText('Insufficient balance')).toBeInTheDocument()
    expect(navigateToConfirmation).not.toHaveBeenCalled()
  })

  it('should navigate to confirmation step when amount is valid', () => {
    const navigateToConfirmation = jest.fn()
    jest.mocked(useParaTimes).mockReturnValue({
      ...mockUseParaTimesResult,
      transactionForm: {
        amount: '1',
      },
    } as ParaTimesHook)
    jest.mocked(useParaTimesNavigation).mockReturnValue({
      ...mockUseParaTimesNavigationResult,
      navigateToConfirmation,
    })
    render(<TransactionAmount />)

    userEvent.click(screen.getByRole('button', { name: 'Next' }))

    expect(navigateToConfirmation).toHaveBeenCalled()
  })

  it('should fetch non EVMc paraTime balance', () => {
    jest.mocked(useParaTimes).mockReturnValue({
      ...mockUseParaTimesResult,
      isDepositing: false,
    } as ParaTimesHook)
    render(<TransactionAmount />)

    expect(mockDispatch).toHaveBeenCalledWith({
      payload: {
        address: 'dummyAddress',
        paraTime: 'cipher',
      },
      type: paraTimesActions.fetchBalanceUsingOasisAddress.type,
    })
  })

  it('should fetch EVMc paraTime balance', () => {
    jest.mocked(useParaTimes).mockReturnValue({
      ...mockUseParaTimesResult,
      isDepositing: false,
      isEvmcParaTime: true,
      transactionForm: {
        ...mockUseParaTimesResult.transactionForm,
        paraTime: ParaTime.Emerald,
      },
    } as ParaTimesHook)
    render(<TransactionAmount />)

    expect(mockDispatch).toHaveBeenCalledWith({
      payload: {
        paraTime: 'emerald',
        privateKey: 'privateKey',
      },
      type: paraTimesActions.fetchBalanceUsingEthPrivateKey.type,
    })
  })
})
