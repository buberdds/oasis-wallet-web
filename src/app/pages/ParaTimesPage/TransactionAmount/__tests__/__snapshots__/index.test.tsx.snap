// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransactionAmount /> should render EVMc withdraw variant component 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  text-align: center;
}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

<span
  class="c0"
  data-testid="paraTime-content-description"
>
  Please enter the amount of ROSE tokens you wish to transfer from the withdrawing wallet on the 
  <strong>
    Emerald
  </strong>
   (EVMc) ParaTime and then click "Next"
</span>
`;

exports[`<TransactionAmount /> should render component 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 48px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 96px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 24px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin: 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 384px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: 24px;
  padding-bottom: 24px;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  font-size: 18px;
  line-height: 24px;
  text-align: center;
}

.c13 {
  font-size: 14px;
  line-height: 20px;
  font-weight: bolder;
}

.c14 {
  margin-left: 3px;
  font-size: 12px;
  line-height: 18px;
  color: lightText;
  font-weight: 600;
}

.c11 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c11:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c11:focus > circle,
.c11:focus > ellipse,
.c11:focus > line,
.c11:focus > path,
.c11:focus > polygon,
.c11:focus > polyline,
.c11:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c11:focus::-moz-focus-inner {
  border: 0;
}

.c11:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c11:focus:not(:focus-visible) > circle,
.c11:focus:not(:focus-visible) > ellipse,
.c11:focus:not(:focus-visible) > line,
.c11:focus:not(:focus-visible) > path,
.c11:focus:not(:focus-visible) > polygon,
.c11:focus:not(:focus-visible) > polyline,
.c11:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c11:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c17 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  margin-bottom: 24px;
  border: 2px solid #7D4CDB;
  border-radius: 18px;
  color: #444444;
  padding: 4px 20px;
  font-size: 14px;
  line-height: 20px;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
}

.c17:hover {
  box-shadow: 0px 0px 0px 2px #7D4CDB;
}

.c17:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c17:focus > circle,
.c17:focus > ellipse,
.c17:focus > line,
.c17:focus > path,
.c17:focus > polygon,
.c17:focus > polyline,
.c17:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c17:focus::-moz-focus-inner {
  border: 0;
}

.c17:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c17:focus:not(:focus-visible) > circle,
.c17:focus:not(:focus-visible) > ellipse,
.c17:focus:not(:focus-visible) > line,
.c17:focus:not(:focus-visible) > path,
.c17:focus:not(:focus-visible) > polygon,
.c17:focus:not(:focus-visible) > polyline,
.c17:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c17:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c21 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #7D4CDB;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  background-color: #7D4CDB;
  color: #f8f8f8;
  border-radius: 18px;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  width: 100%;
}

.c21:hover {
  box-shadow: 0px 0px 0px 2px #7D4CDB;
}

.c21:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c21:focus > circle,
.c21:focus > ellipse,
.c21:focus > line,
.c21:focus > path,
.c21:focus > polygon,
.c21:focus > polyline,
.c21:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c21:focus::-moz-focus-inner {
  border: 0;
}

.c21:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c21:focus:not(:focus-visible) > circle,
.c21:focus:not(:focus-visible) > ellipse,
.c21:focus:not(:focus-visible) > line,
.c21:focus:not(:focus-visible) > path,
.c21:focus:not(:focus-visible) > polygon,
.c21:focus:not(:focus-visible) > polyline,
.c21:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c21:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c19 {
  -webkit-transition: max-height 200ms,opacity 200ms;
  transition: max-height 200ms,opacity 200ms;
  opacity: 0;
  overflow: hidden;
}

.c1 {
  margin-bottom: 48px;
  font-size: 26px;
  line-height: 32px;
  max-width: 624px;
  font-weight: 600;
}

.c10 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  outline: none;
  border: none;
}

.c10::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c10::-moz-placeholder {
  color: #AAAAAA;
}

.c10:-ms-input-placeholder {
  color: #AAAAAA;
}

.c10::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c10::-moz-focus-inner {
  border: none;
  outline: none;
}

.c10:-moz-placeholder,
.c10::-moz-placeholder {
  opacity: 1;
}

.c9 {
  position: relative;
  width: 100%;
}

@media only screen and (max-width:768px) {
  .c2 {
    margin-bottom: 24px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    margin-bottom: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c6 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c7 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c8 {
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c15 {
    padding-top: 12px;
    padding-bottom: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c20 {
    margin-bottom: 24px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    margin-bottom: 24px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    font-size: 18px;
    line-height: 24px;
    max-width: 432px;
  }
}

<div>
  <div
    class="c0"
  >
    <h3
      class="c1"
    >
      Deposit to ParaTime
    </h3>
    <div
      class="c2"
      style="max-width: 550px;"
    >
      <span
        class="c3"
        data-testid="paraTime-content-description"
      >
        Please enter the amount of ROSE tokens you wish to transfer to the receiving wallet on the 
        <strong>
          Cipher
        </strong>
          ParaTime and then click "Next"
      </span>
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        <form
          novalidate=""
        >
          <div
            class="c6"
          >
            <div
              class="c7"
              style="position: relative;"
            >
              <div
                class="c6 "
                style="width: 100%;"
              >
                <div
                  class="c8 FormField__FormFieldContentBox-sc-m9hood-1"
                >
                  <div
                    class="c9"
                  >
                    <input
                      autocomplete="off"
                      class="c10"
                      inputmode="decimal"
                      name="amount"
                      placeholder="0"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <button
                class="c11"
                style="font-weight: bold; z-index: 2; position: absolute; top: 15px; right: 0px;"
                type="button"
              >
                MAX
              </button>
            </div>
            <div
              class="c12"
            >
              <span
                class="c13"
              >
                Available:
                 
                1.0
                <span
                  class="c14"
                />
              </span>
            </div>
          </div>
          <div
            class="c15"
          >
            <div
              class="c16"
            >
              <button
                class="c17"
                type="button"
              >
                Advanced
              </button>
            </div>
            <div
              aria-hidden="true"
              class="c18 c19"
            />
          </div>
          <div
            class="c20"
          >
            <div
              class="c5"
            >
              <button
                class="c21"
                type="submit"
              >
                Next
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
